src/TransactionController.ts

```diff
@@ -354,30 +352,31 @@ export class TransactionController extends BaseController<TransactionConfig, Tra
 	}
 
 	/**
-	 * Resiliently checks all submitted transactions for confirmations in all
-	 * blocks maintained in a sibling BlockHistoryController
+	 * Resiliently checks all submitted transactions on the blockchain
+	 * and verifies that it has been included in a block
+	 * when that happens, the tx status is updated to confirmed
 	 *
 	 * @returns - Promise resolving when this operation completes
 	 */
 	async queryTransactionStatuses() {
 		const { transactions } = this.state;
-		const blockHistory = this.context.BlockHistoryController as BlockHistoryController;
-		const network = this.context.NetworkController as NetworkController;
+		const network = this.context.NetworkController;
 		const currentNetworkID = network.state.network;
-
-		const confirmedHashes = blockHistory.state.recentBlocks
-			.reduce((hashes: string[], block) => {
-				return hashes.concat(block.transactions);
-			}, [])
-			.map((transaction: any) => transaction && transaction.hash);
-
-		transactions.forEach((meta, index) => {
-			const isConfirmed = confirmedHashes.indexOf(meta.transactionHash!) > -1;
-			if (meta.networkID === currentNetworkID && meta.status === 'submitted' && isConfirmed) {
-				transactions[index].status = 'confirmed';
-				this.hub.emit(`${meta.id}:confirmed`, meta);
+		transactions.forEach(async (meta, index) => {
+			if (meta.status === 'submitted' && meta.networkID === currentNetworkID) {
+				try {
+					const txObj = await this.query('getTransactionByHash', [meta.transactionHash]);
+					/* istanbul ignore else */
+					if (txObj && txObj.blockNumber) {
+						transactions[index].status = 'confirmed';
+						this.hub.emit(`${meta.id}:confirmed`, meta);
+					}
+				} catch (e) {
+					/* istanbul ignore next */
```

Comment:
Is this error silencing intentional? If so, you may be able to use the `safelyExecute` helper function.

Reply:
oh cool! I didn't know about that helper